sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1134,3 +1139,110 @@ def convert_row_to_dict(self, row, schema):
       else:
         result[field.name] = self._convert_cell_value_to_dict(value, field)
     return result
+
+
+class BigQueryWriteFn(DoFn):
+  """A ``DoFn`` that streams writes to BigQuery once the table is created.
+  """
+
+  def __init__(self, table_id, dataset_id, project_id, batch_size, schema,
+               create_disposition, write_disposition, client):
+    self.table_id = table_id
+    self.dataset_id = dataset_id
+    self.project_id = project_id
+    self.schema = schema
+    self.client = client
+    self.create_disposition = create_disposition
+    self.write_disposition = write_disposition
+    self._rows_buffer = []
+    self._max_batch_size = batch_size or 500
+
+  def start_bundle(self):
+    self._rows_buffer = []
+    # Transform the table schema into a bigquery.TableSchema instance.
+    if isinstance(self.schema, basestring):
+      # TODO(silviuc): Should add a regex-based validation of the format.
+      table_schema = bigquery.TableSchema()
+      schema_list = [s.strip(' ') for s in self.schema.split(',')]
+      for field_and_type in schema_list:
+        field_name, field_type = field_and_type.split(':')
+        field_schema = bigquery.TableFieldSchema()
+        field_schema.name = field_name
+        field_schema.type = field_type
+        field_schema.mode = 'NULLABLE'
+        table_schema.fields.append(field_schema)
+      self.table_schema = table_schema
+    elif self.schema is None:
+      # TODO(silviuc): Should check that table exists if no schema specified.
+      self.table_schema = self.schema
+    elif isinstance(schema, bigquery.TableSchema):
+      self.table_schema = self.schema
+    else:
+      raise TypeError('Unexpected schema argument: %s.' % self.schema)
+
+    self.bigquery_wrapper = BigQueryWrapper(client=self.client)
+    self.bigquery_wrapper.get_or_create_table(
```

Comment:
yes it does. I was running into issues due to write truncate being a delate and recreate